src/integrations/terminal/TerminalProcess.ts

```diff
@@ -35,6 +44,26 @@ export class TerminalProcess extends EventEmitter<TerminalProcessEvents> {
 			let didOutputNonCommand = false
 			let didEmitEmptyLine = false
 			for await (let data of stream) {
+				// Add to total bytes before checking size
+				const dataBytes = Buffer.from(data).length
+				this.totalBytes += dataBytes
+
+				// Check total accumulated size against half of context limit
+				// Use wouldExceedSizeLimit to avoid creating unnecessary buffer
+				if (wouldExceedSizeLimit(this.totalBytes, this.contextLimit)) {
+					// Create size estimate only when needed for error details
+					const sizeEstimate = estimateContentSize(Buffer.alloc(this.totalBytes), this.contextLimit)
+					this.emit(
+						"error",
+						new ContentTooLargeError({
+							type: "terminal",
+							command,
+							size: sizeEstimate,
+						}),
+					)
+					return
+				}
+
```

Comment:
Would it make sense here to instead of creating an error state, we create a "warn" state, and only return the last portion of the terminal output that would fit in the context limit?

Reply:
Yeah, this should be an error state, because we don't know if the content that gets selected to keep actually holds the useful information the model will need.